### PR TITLE
python2.7-filelock: Init version 3.2.1 for Python 2

### LINKED ISSUE #145109: Nixops does not build due to some of its dependencies no longer supporting Python 2
### Describe the bug
For me, at commit b67e752c29f18a0ca5534a07661366d6a2c2e649 (latest `nixos-unstable` release) and at commit 89ec09c85893fa1c9169aa14509e19e697871f20 (latest in the `staging-next` branch), does not build anymore, due some of its dependencies having no Python 2 support anymore.

As far as I can see, `filelock` dropped Python 2 support since release 3.3.0 (nixpkgs commit e7c94d7f48c2b30c284b11e780bf4b7d196087d0) and later this has been enforced within nixpkgs at commit 877cca04d1f8fb6081808fff16b2d50352a4f6fe. And `filelock` is an indirect dependency of `requests`.

### Steps To Reproduce
1. Set up a simple nixops-shell that pins nixpkgs to a certain version by placing the following files in a directory:
    ```nix
    # pin.nix
    let 
      hostPkgs = import <nixpkgs> {};
      pinnedPkgs = builtins.fetchTarball https://github.com/NixOS/nixpkgs/archive/89ec09c85893fa1c9169aa14509e19e697871f20.tar.gz; 
    in import pinnedPkgs {
      config.allowUnfree = true;
   }
   ```

   ```nix
   # shell.nix
   let
     pkgs = import ./pin.nix;
   in pkgs.stdenv.mkDerivation {
     name = "nixops-env";
     NIX_PATH="nixpkgs=${pkgs.path}";
     buildInputs = [ pkgs.nixops ];
   }
   ```
2. Run `nix-shell` in the current directory.

### Expected behavior
Nixops builds fine and a nix-shell will be entered

### Additional context
Backtrace:

```
error: while evaluating the attribute 'buildInputs' of the derivation 'nixops-env' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/stdenv/generic/make-derivation.nix:205:7:
while evaluating the attribute 'out.outPath' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/customisation.nix:157:13:
while evaluating the attribute 'buildInputs' of the derivation 'nixops-1.7' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/stdenv/generic/make-derivation.nix:205:7:
while evaluating the attribute 'out.outPath' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/customisation.nix:157:13:
while evaluating the attribute 'propagatedBuildInputs' of the derivation 'python2.7-boto-2.49.0' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/stdenv/generic/make-derivation.nix:205:7:
while evaluating the attribute 'out.outPath' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/customisation.nix:157:13:
while evaluating the attribute 'propagatedBuildInputs' of the derivation 'python2.7-requests-2.26.0' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/stdenv/generic/make-derivation.nix:205:7:
while evaluating the attribute 'out.outPath' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/customisation.nix:157:13:
while evaluating the attribute 'propagatedBuildInputs' of the derivation 'python2.7-urllib3-1.26.2' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/stdenv/generic/make-derivation.nix:205:7:
while evaluating the attribute 'dev.outPath' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/customisation.nix:157:13:
while evaluating the attribute 'nativeBuildInputs' of the derivation 'python2.7-cryptography-3.3.2' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/stdenv/generic/make-derivation.nix:205:7:
while evaluating the attribute 'out.outPath' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/customisation.nix:157:13:
while evaluating the attribute 'nativeBuildInputs' of the derivation 'python2.7-hypothesis-4.57.1' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/stdenv/generic/make-derivation.nix:205:7:
while evaluating the attribute 'out.outPath' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/customisation.nix:157:13:
while evaluating the attribute 'nativeBuildInputs' of the derivation 'python2.7-pytest-xdist-1.34.0' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/stdenv/generic/make-derivation.nix:205:7:
while evaluating 'getOutput' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/attrsets.nix:489:23, called from undefined position:
while evaluating 'callPackageWith' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/customisation.nix:117:35, called from /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/top-level/python-packages.nix:2666:14:
while evaluating 'makeOverridable' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/customisation.nix:67:24, called from /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/customisation.nix:121:8:
while evaluating anonymous function at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/development/python-modules/filelock/default.nix:1:1, called from /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/customisation.nix:69:16:
while evaluating 'makeOverridablePythonPackage' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/top-level/python-packages.nix:32:37, called from /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/development/python-modules/filelock/default.nix:9:1:
while evaluating 'makeOverridable' at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/customisation.nix:67:24, called from /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/top-level/python-packages.nix:34:12:
while evaluating anonymous function at /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/pkgs/development/interpreters/python/mk-python-derivation.nix:30:1, called from /nix/store/gw5fin8kbdn165ybp0ywkqi5jnxi0kr3-source/lib/customisation.nix:69:16:
filelock-3.3.0 not supported for interpreter python2.7
```

### Notify maintainers
nixops: @adisbladis @aminechikhaoui @edolstra @rob @domenkozar
filelock: @hyphon81

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[chris@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.14.14-1-default, openSUSE Tumbleweed, noversion`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(root): `"nixpkgs-20.09pre240426.f9567594d5a"`
 - channels(chris): `"nixos-21.11pre308897.e41ba381140, nixpkgs-21.11pre309015.01066edddd7"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: 
  - nixpkgs.nixops
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by HenkKalkwater**
###### Motivation for this change
This fixes nixops, as it still was dependent on filelock for Python 2.
Fixes #145109.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/filelock/3.2.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{ lib, buildPythonPackage, fetchPypi }:
+
+buildPythonPackage rec {
+  pname = "filelock";
+  version = "3.2.0";
```
Comment (thiagokokada) [inline pkgs/development/python-modules/filelock/3.2.nix:5]: Any reason to not build version 3.2.1 instead? https://github.com/tox-dev/py-filelock/releases/tag/3.2.1

Python 2 support was dropped in version 3.3.0: https://github.com/tox-dev/py-filelock/releases/tag/3.3.0

Comment (HenkKalkwater) [reply pkgs/development/python-modules/filelock/3.2.nix:5]: That's an oversight on my side. I'll fix it right now.

Comment (thiagokokada) [reply pkgs/development/python-modules/filelock/3.2.nix]: Please also fix the commit message:

```
python2Modules.filelock: init at version 3.2.1
```

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 145121` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>opae</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>bitbucket-cli</li>
    <li>buttersink</li>
    <li>dir2opus</li>
    <li>euca2ools</li>
    <li>fmbt</li>
    <li>nixops</li>
    <li>split2flac</li>
    <li>tsung</li>
    <li>zabbix-cli</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

File: pkgs/development/python-modules/filelock/3.2.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1qry67zv2pmz8px6wdfbjqv75nmryy2ac7asqgs6q6db2722kpcw";
+  };
+
+  meta = with lib; {
+    homepage = "https://github.com/benediktschmitt/py-filelock";
+    description = "A platform independent file lock for Python";
+    license = licenses.unlicense;
+    maintainers = with maintainers; [ hyphon81 ];
```
Comment (thiagokokada) [inline pkgs/development/python-modules/filelock/3.2.nix:16]: Is this you?

If not, please add yourself or leave it blank.

Comment (HenkKalkwater) [reply pkgs/development/python-modules/filelock/3.2.nix:16]: No, that isn't me, I based this file on an older commit of the `default.nix` in the `filelock` folder and forgot to change this field.

Comment (thiagokokada) [reply pkgs/development/python-modules/filelock/3.2.nix:16]: Can you add yourself or remove it them?

Comment (HenkKalkwater) [reply pkgs/development/python-modules/filelock/3.2.nix]: Sorry, I didn´t  stage the file in git when I amended my last commit. Is fixed now.

File: pkgs/development/python-modules/filelock/3.2.nix [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1qry67zv2pmz8px6wdfbjqv75nmryy2ac7asqgs6q6db2722kpcw";
+  };
+
+  meta = with lib; {
+    homepage = "https://github.com/benediktschmitt/py-filelock";
+    description = "A platform independent file lock for Python";
+    license = licenses.unlicense;
+    maintainers = with maintainers; [ henkkalkwater ];
```
Comment (thiagokokada) [inline pkgs/development/python-modules/filelock/3.2.nix:16]: You need to add yourself on `maintainers-list.nix` file.

Comment (HenkKalkwater) [reply pkgs/development/python-modules/filelock/3.2.nix:16]: Sorry,  I wasn't aware of that. Anyhow, thank you for your patience while guiding me through this PR.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 145121` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>opae</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>bitbucket-cli</li>
    <li>buttersink</li>
    <li>dir2opus</li>
    <li>euca2ools</li>
    <li>fmbt</li>
    <li>nixops</li>
    <li>split2flac</li>
    <li>tsung</li>
    <li>zabbix-cli</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Comment by thiagokokada** [comment]
Thanks!

**Pull request merged**